### PR TITLE
[Feature] institutions auth

### PR DESCRIPTION
**Pull request by chrisseto**

### CODE REVIEW

File: api/institutions/views.py [hunk 1]
```diff
@@ -184,3 +188,18 @@ def get_default_odm_query(self):
...
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+    required_read_scopes = [CoreScopes.NULL]
+    required_write_scopes = [CoreScopes.NULL]
+    authentication_classes = (InstitutionAuthentication, )
+    view_category = 'institutions'
+    view_name = 'institution-auth'
+
+    def post(self, request, *args, **kwargs):
```
Comment (hmoco) [inline api/institutions/views.py]: do we ever want to return the serialized user (created or got)?

File: api/institutions/authentication.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        except (jwt.InvalidTokenError, KeyError, ValueError, TypeError):
+            raise AuthenticationFailed
+
+        username = payload['sub']
+        data = json.loads(payload['data'])
+
+        institution = find_institution_by_domain(username)
+        fullname = data.get(institution.metadata_request_fields.get('fullname'))
+
+        user, _ = get_or_create_user(fullname, username)
```
Comment (hmoco) [inline api/institutions/authentication.py]: Using that ` get_or_create_user ` requires sending of signal (to subscribe user to mailchimp). Adding something like:
```
if _:
    signals.user_confirmed.send(user)
```

File: api_tests/institutions/views/test_institution_auth.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+        res = self.app.post(self.url, self.build_payload(username))
+        assert_equal(res.status_code, 204)
+
+        user.reload()
+        assert_equal(len(user.affiliated_institutions), 1)
+
+    def test_bad_token(self):
+        res = self.app.post(self.url, 'al;kjasdfljadf', expect_errors=True)
+        assert_equal(res.status_code, 403)
```
Comment (hmoco) [inline api_tests/institutions/views/test_institution_auth.py]: Good tests, maybe add one that tests the update of a users ` institution_metadata `

**Comment by icereval** [comment]
Updated w/ #5051

**Pull request closed without merging**